src/plugins/content_management/public/services/content_management/page.ts

```diff
@@ -31,6 +35,28 @@ export class Page {
     return this.sections$;
   }
 
+  updateSectionInput(
+    sectionId: string,
+    callback: (section: Section | null, err?: Error) => Section | null
+  ) {
+    const section = this.sections.get(sectionId);
+    if (!section) {
+      callback(null, new Error(`Section id ${sectionId} not found`));
+      return;
+    }
+
+    const updated = callback(section);
+    if (updated) {
```

Comment:
How would you feel about doing
```
if (updated.kind !== section.kind) return;
```
... and then a switch-case for `dashboard`, `card`, and `custom`?

Reply:
Good idea! I will adopt this change in the upcoming PR

Reply:
<img width="1284" alt="image" src="https://github.com/user-attachments/assets/cd4a2171-1a9c-44e5-85e8-eb5575cc04a7">
Seems TS compiler is not happy with that ;D